spec/datadog/profiling/collectors/cpu_and_wall_time_worker_spec.rb

```diff
@@ -897,6 +914,78 @@
         # 40 is the size of a basic object and we have test_num_allocated_object of them
         expect(total_size).to eq test_num_allocated_object * 40
       end
+
+      describe "heap cleanup after GC" do
+        let(:options) { {dynamic_sampling_rate_enabled: false} }
+
+        let(:cleared_object_id) do
+          stub_const("CpuAndWallTimeWorkerSpec::TestStruct", Struct.new(:foo))
+
+          start
+
+          # Force a full GC to make sure there's no incremental GC going on at this point
+          GC.start
+
+          test_object = CpuAndWallTimeWorkerSpec::TestStruct.new
+          test_object_id = test_object.object_id
+
+          expect(
+            Datadog::Profiling::StackRecorder::Testing._native_is_object_recorded?(recorder, test_object_id)
+          ).to be true
+
+          # Let's replace the test_object reference with another object, so that the original one can be GC'd
+          test_object = Object.new # rubocop:disable Lint/UselessAssignment
+
+          # Force an update to happen on the next GC
+          Datadog::Profiling::StackRecorder::Testing._native_heap_recorder_reset_last_update(recorder)
+
+          GC.start
+
+          test_object_id
```

Comment:
From previous flakes we may need to redo this a bunch of times just in case we have a ref in a register somewhere keeping it alive?

Reply:
Yeah, I wondered about that as well.

In this case I'm explicitly allocating a new object + changing the local variable + and this is in a separate method which was somewhat deliberate as "a bunch of random actions to hopefully flush any hidden/leftover references".

I guess we'll see if we get lucky or not?